### ISSUE TITLE
Use service logon credentials for SMB deployer

### ISSUE THREAD

**Issue by dassie** [JetBrains/teamcity-deployer-plugin#67]
Is there a way to have the SMB deployer use the teamcity service's logon credentials instead of having to enter a username and password in the step? Similar 

The problem isn't that it's tedious, it's that it's not really safe to store a password in plaintext.

**Comment by nskvortsov** [comment]
Hello David,

Currently, there is no such feature. While it sounds like a good feature
request, please note, that artifacts are published from TeamCity Build
Agent (not TeamCity Server). So you will need to make sure, that each agent
on your grid runs under correct credentials.

**Comment by dassie** [comment]
Yeah, that was my bad. I meant the agent service logon credentials. But you're right, I would need to insure that all agents have access to the file share. My team uses a single account for all the agents and the teamcity service. Not sure how common that is.

I tried to use smb deployer to get rid of some powershell scripts I wrote to upload artifacts. But the user/password text boxes were a deal breaker unfortunately. :(
